src/main/java/org/spongepowered/api/plugin/Plugin.java

```diff
@@ -24,19 +24,21 @@
  */
 package org.spongepowered.api.plugin;
 
-import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
+import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import java.util.regex.Pattern;
 
 /**
  * An annotation used to describe and mark a Sponge plugin.
  */
-@Target(TYPE)
-@Retention(RUNTIME)
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
 public @interface Plugin {
 
+    Pattern ID_PATTERN = Pattern.compile("[a-z0-9-_]+");
```

Comment:
Personally I feel that if there's going to be some large fraction of the amount of Bukkit plugins there are currently but for Sponge in the future, we'll need to be able to support `.` separated plugin id's at the least. Especially with all sorts of library plugins etc that will be introduced. As @Mumfrey said, it's best practice to encourage good practices. I'd vote for plugin id's being as free of restriction as possible to allow true uniqueness for plugin developers to identify their own creations compared to others.